pegasus/helpers/section_api_helpers.rb

```diff
@@ -193,6 +193,53 @@ def self.encrypt_password(password)
   end
 end
 
+class DashboardCourseExperiments
+  # Fetches a list of all experiments pertaining to courses and caches the
+  # result.
+  # @return [Array[String] An array of experiment names.
+  @@course_experiments = nil
+  def self.course_experiments
+    @@course_experiments ||= Dashboard.db[:course_scripts].
+      exclude(experiment_name: nil).
+      all.
+      map {|cs| cs[:experiment_name]}
+  end
+
+  # Fetches course experiment data from the dashboard DB and returns
+  # a map such that map[user_id][experiment_name] is true if the user
+  # belongs to the specified experiment. Caches the map so that the
+  # database fetch is only done once per frontend.
+  # @return [Hash[Hash[Boolean]]] A 2-dimensional map from user id and
+  # experiment name to boolean.
+  @@course_experiment_map = nil
+  def self.course_experiment_map
+    @@course_experiment_map ||= {}.tap do |map|
+      Dashboard.db[:experiments].
+        where(name: course_experiments, type: 'SingleUserExperiment').
+        all.
+        each do |row|
+        user_id = row[:min_user_id]
+        map[user_id] ||= {}
+        map[user_id][row[:name]] = true
+      end
+    end
+  end
```

Comment:
A per user_id cache seems like it could get pretty large. Do we have a sense of approx how many users we might have in this cache?

Reply:
We will have `O(100)` users with course experiments over the next 3-6 months.

Reply:
Are you planning on only using `SingleUserExperiment`? I thought you wanted to turn it on for O(100) whole sections.

Reply:
I am planning to turn this on for O(100) sections, however it will be changed to use a new experiment type in the next PR. From the dev doc:

> create a new SingleTeacherExperiment which borrows aspects of SingleUserExperiment and TeacherBasedExperiment. An instance of this enables the experiment for a teacher and their students. The experiment flag will only ever be stored on the teacher, so the scale will be O(100) new entries in the underlying experiments table, which currently has ~600 entries. This is important because all entries in the experiments table are cached on every frontend.